content/docs/command-reference/add.md

```diff
@@ -174,8 +175,13 @@ not.
   when using `--to-remote`. The default value is `4 \* cpu_count()`. For SSH
   remotes, the default is `4`. Using more jobs may speed up the operation.
 
-- `--desc <text>` - user description of the data (optional). This doesn't affect
-  any DVC operations.
+- `--desc <text>` - user description of the data.
+
+- `--type <str>` - user assigned type of the data.
+
+- `--label <text>` - user assigned label(s) to add to the data.
+
+- `--meta key=value` - custom metadata to add to the data.
```

Comment:
Product questions

- I think they're still optional? Probably worth mentioning
  Maybe having said optional the "user" wording is no longer necessary: I don't think it's clear, as everything in the pipeline is defined by the user.
- Why are some `<text>` and some `<str>`? Distinction unclear (or needs more detail)
- Should we use `<>` around `key=value`?
- "metadata ... to the data" seems redundant, can we rephrase?

Syntax-wise

- Pls hyphenate "user-assigned" if we keep that term.
- No periods `.` needed (not full sentences).

Thanks, I'll leave the rest here to the DVC team as it's mainly a technical reference type contribution! But please @ mention me if needed.

Reply:
> * I think they're still optional? Probably worth mentioning

Just for clarification: almost all of the CLI flags are optional, so I questioned why this was needed in the help text of the command. However, agree that we should mention it in the dvc.yaml reference fields where we do otherwise specify what is optional.

Reply:
In dvc.yaml too, everything is optional except for `cmd`.

Reply:
In `.dvc` file, everything is optional.

Reply:
Good point. I saw it was mentioned for other fields so was aiming for consistency. Can we remove `(Optional)` in the docs for the other fields where it's mentioned?

@jorgeorpinel Any objections?

Reply:
Makes sense! Maybe just indicate "required" in the few metafile fields/ cmd args/ fn params that are. Goes beyond the scope of this PR though

Reply:
I think for consistency, it makes sense as part of this PR scope.

Reply:
> Maybe having said optional the "user" wording is no longer necessary: I don't think it's clear, as everything in the pipeline is defined by the user.

- `user` is important here. Other keywords have certain meanings or is associated with something. These are however not. Besides, `desc` already has that wording.

> Why are some <text> and some <str>? Distinction unclear (or needs more detail)

It's fine to leave this out. `text` is multiline string. It could be `<label>`, but since we don't have understanding of those metavars, I think it's fine to keep just `<str>`.

> Should we use <> around key=value?

I don't think it's necessary. It should be `<key>=<value>` actually, if we typed it out. But simple is better, no need to complicate here.

> "metadata ... to the data" seems redundant, can we rephrase?

What do you propose?

Pls hyphenate "user-assigned" if we keep that term.

Sure, I can do that.

> No periods . needed (not full sentences).

We already have some phrases/sentences with periods. It's better to be consistent. I am not fixing it at this time.

Reply:
Let's update in a separate PR. It touches other parts too, eg: `stage` definition.

Reply:
Sounds good, @skshetry . Could you make those changes?

Reply:
I added to this PR to save from having to make another one.